### PR TITLE
svg fill-rule acknowledgement 

### PR DESCRIPTION
**Pull request by jpirnay**
Acknowledge fill-rule attribute:
image
Should fix [Issue 1020](https://github.com/meerk40t/meerk40t/issues/1020)

### CODE REVIEW

File: meerk40t/core/node/elem_path.py [hunk 1]
```diff
@@ -136,4 +141,5 @@ def as_path(self):
         self.path.stroke_width = self.stroke_width
         self.path.linecap = self.linecap
         self.path.linejoin = self.linejoin
+        self.shape.fillrule = self.fillrule
```
Comment (tatarize) [inline meerk40t/core/node/elem_path.py:144]: This really should be `self.path` elem_path doesn't have a self.shape value.

File: meerk40t/core/node/elem_line.py [hunk 2]
```diff
@@ -136,4 +141,5 @@ def as_path(self):
         self.shape.stroke_width = self.stroke_width
         self.shape.linecap = self.linecap
         self.shape.linejoin = self.linejoin
+        self.shape.fillrule = self.fillrule
```
Comment (tatarize) [inline meerk40t/core/node/elem_line.py:144]: I guess this is fine, lines are never filled though, but for consistency I guess... but if that's the case, why not rect and circle? (those are non-crossing and can't need a fill rule either).

**Review by tatarize — CHANGES_REQUESTED** [review]
Small bug in path (should crash), and some inconsistency with line but not ellipse/rect.

**Comment by tatarize** [comment]
Fixes #1020

**Comment by tatarize** [comment]
Also it's not really too essential there. The shape linecap and linejoin aren't actually part of `svgelements` so writing them doesn't matter.

In the case with stroke_width and the path it does matter since that code is still used for bounds checking here and there since it can calculate the bounds of a some path. And this can include or exclude the stroke width. This matters for cases when you're rendering them to raster them. So in cases when you want to know this information you set these back on the path so that the calculations work.

The linecap and linejoin and fillrule aren't respected by the `svgelements` since there's nothing about them that matters. In theory, if I get a free month or couple weeks and fully implement my `svgio` project these things would exist as `@property` values, and thus could matter though it would set clear boundaries between rendered paths and `<Path>` node objects, which `svgelements` does not do much to it's determent.

**Comment by jpirnay** [comment]
I've added fillrule to those elementd where they are not actually needed per se. Not sure how the SVG-standard deals with those attributes for rects and ellipses... So this may need to be revisited

**Review by tatarize — APPROVED** [review]
The fill rules won't matter for them since they by definition simple. But, writing that stuff doesn't matter that much.

**Pull request merged**